File: PromptCraft-Studio/backend/src/main/java/com/promptcraft/entity/Project.java
package com.promptcraft.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonRawValue;
import lombok.*;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@TableName("project")
public class Project {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    /** Stored as a JSON array string in MySQL JSON column, emitted as raw JSON in responses */
    @TableField("tags")
    @JsonRawValue
    private String tags;

    private Long stableVersionId;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: PromptCraft-Studio/frontend/src/types/project.ts
export interface Project {
  id: number
  name: string
  description: string
  tags: string[]
  stableVersionId: number | null
  createdAt: string
  updatedAt: string
}

export interface ProjectVO extends Project {
  versionCount: number
  stableVersionNo?: number
}

export interface ProjectDTO {
  name: string
  description: string
  tags: string[]
}

export interface ProjectFilter {
  keyword?: string
  tags?: string[]
}
